elf/io/files.py

```diff
@@ -26,6 +27,11 @@ def open_file(path, mode='a', ext=None):
         ext [str] - file extension. This can be used to force an extension
             if it cannot be inferred from the filename. (default: None)
     """
+    # Before checking the extension suffix, check for "protocol-style"
+    # cloud provider prefixes.
```

Comment:
I chose to do it this way so that everything fits nicely in the existing extension framework; but we could also have `CLOUD_CONSTRUCTORS` totally separately. I don't think it's necessary, and probably breaks things downstream; but it's slightly more explicit.

Reply:
I like this solution, fits the current code well :+1: